src/Form/Field.php

```diff
@@ -213,47 +213,43 @@ public static function defaults(): array
             'computed' => [
                 'after' => function () {
                     /** @var \Kirby\Form\Field $this */
-                    if ($this->after !== null) {
-                        return $this->model()->toString($this->after);
-                    }
+                    return $this->toString($this->after);
                 },
                 'before' => function () {
                     /** @var \Kirby\Form\Field $this */
-                    if ($this->before !== null) {
-                        return $this->model()->toString($this->before);
-                    }
+                    return $this->toString($this->before);
                 },
                 'default' => function () {
                     /** @var \Kirby\Form\Field $this */
-                    if ($this->default === null) {
-                        return;
-                    }
-
-                    if (is_string($this->default) === false) {
-                        return $this->default;
-                    }
-
-                    return $this->model()->toString($this->default);
+                    return $this->toString($this->default);
                 },
                 'help' => function () {
                     /** @var \Kirby\Form\Field $this */
-                    if ($this->help) {
-                        $help = $this->model()->toString($this->help);
-                        $help = $this->kirby()->kirbytext($help);
-                        return $help;
+                    if ($text = $this->toString($this->help)) {
+                        return $this->kirby()->kirbytext($text);
                     }
                 },
                 'label' => function () {
                     /** @var \Kirby\Form\Field $this */
-                    if ($this->label !== null) {
-                        return $this->model()->toString($this->label);
-                    }
+                    return $this->toString($this->label);
                 },
                 'placeholder' => function () {
                     /** @var \Kirby\Form\Field $this */
-                    if ($this->placeholder !== null) {
-                        return $this->model()->toString($this->placeholder);
+                    return $this->toString($this->placeholder);
+                }
+            ],
+            'methods' => [
+                'toString' => function ($value = null) {
```

Comment:
We can move, but will this method;

- work for the field value?

````php
public function toString()
{
    $value = $field->value;
    //
}
````

- or as an helper method for passing values?


````php
public function toString($value = null)
{
    //
}
````

It's a little confusing 🤔

Reply:
A few comments:

- What would be the use-case for passing `null` as the `$template`? Or is that just for compatibility to avoid fatal errors if `null` is passed through? If so: Can we safely assume that every use of this method can deal with a returned `null` value?
- If we decide to allow `null`, the argument should be hinted `?string $template = null`
- If the first argument is hinted as `?string $template = null`, it can never be anything else than `null` or `string`, so the second `if` doesn't do anything.

Reply:
> If the `toString()` method doesn't return `null` (because the value of the undefined prop is `null`), the props of all fields will be empty string (`''`) instead of `null` (For ex: `before`, `after`, `default`, etc..). This is undesirable.

Very good point, I fully agree!